### PR TITLE
RavenDB-5901 - Fetch the stats of the database when clicking on its block

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-5901

### CODE REVIEW

File: src/Raven.Studio/wwwroot/App/views/resources/resources.html [hunk 1]
```diff
@@ -78,7 +78,7 @@
     <div data-bind="foreach: resources().sortedResources">
         <div class="panel resource" data-bind="scrollTo: isCurrentlyActiveResource(), visible: !filteredOut(), attr: { class: 'panel resource ' + badgeClass() + ' ' + (isCurrentlyActiveResource() ? 'active' : '') }">
             <div class="padding padding-sm">
-                <div data-bind="attr: { 'data-state-text': badgeText, class: 'state ' + badgeClass() }"></div>
+                <div style="cursor: pointer" data-bind="click: $root.activateResource, attr: { 'data-state-text': badgeText, class: 'state ' + badgeClass() }"></div>
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/resources/resources.html:81]: Can we use less files for this?

File: src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts [hunk 2]
```diff
@@ -44,6 +44,34 @@ class databaseInfo extends resourceInfo {
         this.documentsCount(databaseInfo.DocumentsCount);
         this.indexesCount(databaseInfo.IndexesCount);
     }
+
+    static empty(resourceName: string): databaseInfo {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts:48]: Can we avoid having this method? Instead when we get info that new resource was created, then fetch information only for this resource and then add to resources list? I think @Danielle9897 so implementing something similar.

File: src/Raven.Studio/typescript/viewmodels/resources/resources.ts [hunk 3]
```diff
@@ -348,6 +371,20 @@ class resources extends viewModelBase {
         app.showBootstrapDialog(createDbView);
     }
 
+    activateResource(rsInfo: resourceInfo) {
+        let resource = this.resourcesManager.resources().filter(x => x.qualifiedName === rsInfo.qualifiedName)[0];
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/resources/resources.ts:375]: use find instead.

File: src/Raven.Studio/typescript/viewmodels/resources/resources.ts [hunk 4]
```diff
@@ -348,6 +371,20 @@ class resources extends viewModelBase {
...
+        if (!resource)
+            return;
+
+        activeResourceTracker.default.resource(resource);
+        resource.activate();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/resources/resources.ts:380]: this call should trigger EVENT.Resource.Activate and update activeResourceTracker.default.resource(resource);, so please remove previous line.

**Review by ml054 — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts:48]: We cannot get the information about the database until it's actually loaded anyway.
After we receive the "Load" notification we fetch the stats.

File: src/Raven.Studio/typescript/viewmodels/resources/resources.ts (hunk 4 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/viewmodels/resources/resources.ts:380]: Without this line there is a delay in the resource activation.
I'll see why

File: src/Raven.Studio/wwwroot/App/views/resources/resources.html (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/resources/resources.html:81]: Yes

File: src/Raven.Studio/typescript/viewmodels/resources/resources.ts (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/viewmodels/resources/resources.ts:375]: ok

File: src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts (hunk 2 above)
Comment (ml054) [reply src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts:48]: not necessarily it's true.  tal implemented cache so we can fetch info even about offline databases, but event had name loaded not load  ( changed few days ago by Danielle) so right now we have assurance that database is already loaded when this notification arrives. so we remove this empty method and load info and then display on screen

Comment (grisha-kotler) [reply src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts:48]: What if I changed the database document (which will trigger a "Write")? This shouldn't "wake up" the database

Comment (ml054) [reply src/Raven.Studio/typescript/models/resources/info/databaseInfo.ts:48]: It shouldn't:
In general we should call /resources?name={dbName}&type=db
but this call can't wake database up ( and I have hope it is not). 

In v3.5 we fetched information using /stats endpoint, and it was waken up database. In v4.0 we have different behavior.

**Comment by ayende** [comment]
Please open a new one with the updated changes

**Pull request closed without merging**